engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -195,6 +199,21 @@ Stateful doPolyglot(
       }
     }
     Object res = hostMethodCallNode.execute(polyglotCallType, symbol.getName(), self, args);
+    if (!(res instanceof Atom)) {
+      if (interop.isDate(res)) {
```

Comment:
This is horrid, why do we need to wrap everything so early? Why can't a date just flow through the system normally?

Reply:
Quite convincing @kustosz! You'll get `EnsoDate` then: [93abfa1](https://github.com/enso-org/enso/pull/3559/commits/93abfa1644637c08677f844d3953ad153e1dd19b)